Remove overloads from tutorial on content negotiation (#2682)
microsoft/typespec: docs/standard-library/http/content-types.md

## Content type negotiation

> There could be cases where you might the same endpoint to return different content depending on the content type requested. To achieve this scenario, you need to use using shared routes where different content response is represented as a different operation that share the same endpoint.

Reviewer: ```suggestion
There could be cases where the same endpoint returns different content depending on the content type requested. To achieve this scenario, use shared routes with each different content response is represented as a different operation that shares the same endpoint.
```

@sharedRoute
op getAvatarAsJpeg(@header accept: "image/jpeg"): JpegImage;
```

> ### Option 2: Using overload

Reviewer: Not sure why we are removing this - it should be possible for folks to use this mechanism, even though it is not supported in DPG emitters

Reviewer 2: yeah lets discuss that in the dpg meeting, maybe what we want is a linter rule for azure spec not to use `@overload` or we can discusss if we want to stop supporting it all together but I don't think we should just remove it from the public doc. Azure docs should already have similar docs with the resource templates that only show the sharedRoute flow
